### PR TITLE
update readme to remove residual binary references

### PR DESCRIPTION
**Pull request by sjwiesman**
### Motivation

Updates the README to remove residual binary references. We need to do a larger rewrite, but this is hopefully good enough for tomorrow.

### CODE REVIEW

**Review by morsapaes — APPROVED** [review]
Pushed in a couple of suggestions! And agreed that both this and the overview pages in the documentation need a complete rewrite that follows our updated positioning. I kind of want to drop the architecture diagram in here, but also know that @benesch isn't the biggest fan.

**Comment by sjwiesman** [comment]
Love the suggestions. I’d vote to merge

**Review by frankmcsherry — APPROVED** [review]
All of these changes seem fine to me! @benesch would be best to confirm, but .. I'm game!

File: README.md [hunk 1]
```diff
@@ -4,25 +4,23 @@
...
 
-Materialize focuses on providing correct and consistent answers with minimal latency. It does not ask you to accept either approximate answers or eventual consistency. Whenever Materialize answers a query, that answer is the correct result on some specific (and recent) version of your data. Materialize does all of this by recasting your SQL92 queries as *dataflows*, which can react efficiently to changes in your data as they happen. Materialize is powered by [timely dataflow](https://github.com/TimelyDataflow/timely-dataflow), which connects the times at which your inputs change with the times of answers reported back to you.
+It focuses on providing correct and [consistent](https://materialize.com/docs/overview/isolation-level/) answers with minimal latency, and does not ask you to accept either approximate answers or eventual consistency. Whenever Materialize answers a query, that answer is the correct result on some specific (and recent) version of your data. Materialize does all of this by recasting your SQL92 queries as *dataflows*, which can react efficiently to changes in your data as they happen.
```
Comment (benesch) [inline README.md:17]: Can we ditch the “SQL92” here and just say “SQL”? I doubt we support everything in SQL 92, and also we support plenty of things outside of SQL 92.

**Review by benesch — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -103,10 +108,6 @@ Materialize is source-available and [licensed](LICENSE) under the BSL 1.1, conve
 
 Materialize is also available as [a paid cloud service](https://materialize.com/pricing/) with additional features such as high availability via multi-active replication.
 
-## How does it work?
-
-Materialize is built on top of [differential dataflow](https://github.com/TimelyDataflow/differential-dataflow) and [timely dataflow](https://github.com/TimelyDataflow/timely-dataflow), and builds on a decade of cutting-edge stream processing research.
-
 ## For developers
 
 Materialize is written entirely in Rust.
```
Comment (benesch) [inline README.md:113]: This isn’t entirely true either! It has a little bit of C.

**Review by benesch — COMMENTED** [review]
I’m happy to address the two things I just mentioned in a follow up.

File: README.md (hunk 2 above)
Comment (morsapaes) [reply README.md]: Changed it to "primarily written in Rust".

**Comment by morsapaes** [comment]
Thanks, @frankmcsherry @benesch! Like Seth mentioned, this is ripe for a full rewrite, so I'll merge it as is and make sure a follow-up PR comes through.

**Pull request merged**